zerver/tests/tests.py

```diff
@@ -986,6 +986,54 @@ def test_notify_settings(self):
         self.assertEqual(get_user_profile_by_email("hamlet@zulip.com").
                 enable_desktop_notifications, False)
 
+        json_result = self.client.post("/json/notify_settings/change",
+                                       {"enable_stream_desktop_notifications": ujson.dumps(False)})
+        self.assert_json_success(json_result)
+        self.assertEqual(get_user_profile_by_email("hamlet@zulip.com").
+                enable_stream_desktop_notifications, False)
```

Comment:
can you do this as one test per notify_settings bit feature, each case turning the feature on and off?  

The cleanest way to do this would be to write a test helper function that takes as an argument the name of the field being tested (e.g. in this case, "enable_stream_desktop_notifications"; for the last line of this block you can use getattr() to validate the value.  

And then each test case would call that function, passing a different option.